.github/actions/test/action.yml

```diff
@@ -58,8 +58,8 @@ runs:
       run: tar --exclude-vcs -acf ${{ runner.temp }}/custom.tgz .
 
     - name: Upload custom artifact
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@c7d193f32edcb7bfad88892161225aeda64e9392 # v4
       with:
-        name: custom-${{ github.event.pull_request.head.sha || github.event.head_commit.id }}-${{ runner.os }}-${{ runner.arch }}
+        name: custom-${{ github.event.pull_request.head.sha || github.event.head_commit.id }}-${{ strategy.job-index }}-${{ runner.os }}-${{ runner.arch }}
```

Comment:
you should omit `${{ strategy.job-index }}` from here because it won't necessarily align with the downloaded index